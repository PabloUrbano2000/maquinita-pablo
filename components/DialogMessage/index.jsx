import React from 'react';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';
import Slide from '@mui/material/Slide';

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
});

const DialogMessage = ({ show, setShow, title, message }) => {
  const handleClose = () => {
    setShow(false);
  };

  return (
    <Dialog
      open={show}
      TransitionComponent={Transition}
      keepMounted
      onClose={handleClose}
      aria-describedby="alert-dialog-slide"
    >
      <DialogTitle>{title}</DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-slide">{message}</DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button
          variant="contained"
          style={{ fontSize: '1em', marginTop: 20, marginBottom: 10 }}
          onClick={handleClose}
        >
          Aceptar
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default DialogMessage;
