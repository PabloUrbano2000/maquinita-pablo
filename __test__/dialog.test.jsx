import { render } from '@testing-library/react';
import DialogMessage from '../components/DialogMessage';

it('render component DialogMessage', () => {
  const { container } = render(
    <DialogMessage
      show={false}
      setShow={() => console.log('Set value')}
      title={'Titulo'}
      message={'Mensaje en body'}
    />
  );
  expect(container).toMatchSnapshot();
});
