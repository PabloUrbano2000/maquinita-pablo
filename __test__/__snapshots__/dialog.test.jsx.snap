// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component DialogMessage 1`] = `<div />`;
